src/ExchangeRate.java

```diff
@@ -0,0 +1,219 @@
+
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+/**
+ * {@code ExchangeRate} represented as a {@code Double} with implementations of
+ * primary methods.
+ *
+ * @convention <pre>
+ * [all characters of $this.rep are '0' through '9']
+ * </pre>
+ * @correspondence <pre>
+ * this = [the decimal number whose ordinary depiction is $this.rep]
+ * </pre>
+ *
+ * @author Ryan Ouimet
+ *
+ */
+
+public class ExchangeRate {
+
+    /*
+     * Private members --------------------------------------------------------
+     */
+
+    /**
+     * Representation of {@code this}.
+     */
+    private Double rep;
+
+    /**
+     * Creator of initial representation.
+     */
+    private void createNewRep() {
+        this.rep = new Double(0);
+    }
+
+    /*
+     * Constructors -----------------------------------------------------------
+     */
+
+    /**
+     * No-argument constructor.
+     */
+    public ExchangeRate() {
+        this.createNewRep();
+    }
+
+    /**
+     * Constructor from {@code double}.
+     *
+     * @param d
+     *            {@code double} to initialize from
+     */
+    public ExchangeRate(double d) {
+        this.rep = new Double(d);
+    }
+
+    /**
+     * Constructor from {@code int}.
+     *
+     * @param i
+     *            {@code int} to initialize from
+     */
+    public ExchangeRate(int i) {
+        this.rep = new Double(i);
+    }
+
+    /**
+     * Constructor from {@code String}.
+     *
+     * @param s
+     *            {@code String} to initialize from
+     */
+    public ExchangeRate(String s) {
+        assert s != null : "Violation of: s is not null";
+        assert s.matches(
+                "-?\\d+(\\.\\d{1,2})?") : "Violation of: s must be a double value";
+
+        this.rep = new Double(s);
+    }
+
+    /*
+     * Kernel methods ---------------------------------------------------------
+     */
+
+    /**
+     * Sets an exchange rate based on the ratio of currency2 to currency1.
+     *
+     * @param currency1
+     *            the currency in the denominator
+     *
+     * @param currency2
+     *            the currency in the numerator
+     *
+     */
+    private void setRate(double currency1, double currency2) {
```

Comment:
Does it make sense to have a separate method for this? Could this just be another constructor?